angr/analyses/reaching_definitions/live_definitions.py

```diff
@@ -92,49 +94,51 @@ def _set_initialization_values(self, subject, rtoc_value=None):
             if func_addr is None:
                 l.warning("func_addr must not be None to initialize a function in mips")
-            t9 = Register(self.arch.registers['t9'][0],self.arch.bytes)
+            t9 = Register(self.arch.registers['t9'][0], self.arch.bytes)
             t9_def = Definition(t9, ExternalCodeLocation(), DataSet(func_addr,self.arch.bits))
             self.register_definitions.set_object(t9_def.offset,t9_def,t9_def.size)
 
-        if cc is not None:
+        if cc is not None and cc.args is not None:
             for arg in cc.args:
                 # initialize register parameters
-                if type(arg) is SimRegArg:
+                if isinstance(arg, SimRegArg):
                     # FIXME: implement reg_offset handling in SimRegArg
                     reg_offset = self.arch.registers[arg.reg_name][0]
                     reg = Register(reg_offset, self.arch.bytes)
-                    reg_def = Definition(reg, ExternalCodeLocation(), DataSet(Parameter(reg), self.arch.bits))
+                    reg_def = Definition(reg, ExternalCodeLocation(), DataSet(undefined, self.arch.bits))
                     self.register_definitions.set_object(reg.reg_offset, reg_def, reg.size)
                 # initialize stack parameters
-                elif type(arg) is SimStackArg:
+                elif isinstance(arg, SimStackArg):
                     ml = MemoryLocation(self.arch.initial_sp + arg.stack_offset, self.arch.bytes)
                     sp_offset = SpOffset(arg.size * 8, arg.stack_offset)
-                    ml_def = Definition(ml, ExternalCodeLocation(), DataSet(Parameter(sp_offset), self.arch.bits))
+                    ml_def = Definition(ml, ExternalCodeLocation(), DataSet(undefined, self.arch.bits))
                     self.memory_definitions.set_object(ml.addr, ml_def, ml.size)
                 else:
                     raise TypeError('Unsupported parameter type %s.' % type(arg).__name__)
 
         # architecture dependent initialization
         if self.arch.name.lower().find('ppc64') > -1:
+            if rtoc_value is None:
+                raise TypeError("rtoc_value must be provided on PPC64.")
             offset, size = self.arch.registers['rtoc']
             rtoc = Register(offset, size)
             rtoc_def = Definition(rtoc, ExternalCodeLocation(), DataSet(rtoc_value, self.arch.bits))
             self.register_definitions.set_object(rtoc.reg_offset, rtoc_def, rtoc.size)
         elif self.arch.name.lower().find('mips64') > -1:
             offset, size = self.arch.registers['t9']
             t9 = Register(offset, size)
-            t9_def = Definition(t9, ExternalCodeLocation(), DataSet(func_addr, self.arch.bits))
+            t9_def = Definition(t9, ExternalCodeLocation(), DataSet({func_addr}, self.arch.bits))
             self.register_definitions.set_object(t9.reg_offset, t9_def, t9.size)
 
-    def copy(self):
+    def copy(self) -> 'LiveDefinitions':
```

Comment:
Are the quotes a mistake, or do they have a special meaning?

Reply:
Python (<3.8) will complain since `LiveDefinitions` is the current class that this method belongs to, and it is not considered a full class that can be referenced. I believe this problem is fixed in Python 3.9.